### PR TITLE
 modify write function return -1 when not authorized && fix authority for create table

### PR DESCRIPTION
**Pull request by fqliao**

### CODE REVIEW

File: libstoragestate/StorageState.cpp [hunk 1]
```diff
@@ -400,11 +401,15 @@ bool StorageState::checkAuthority(Address const& _origin, Address const& _contra
...
+void StorageState::createAccount(
+    Address const& _origin, Address const& _address, u256 const& _nonce, u256 const& _amount)
 {
     std::string tableName("_contract_data_" + _address.hex() + "_");
-    auto table = m_memoryTableFactory->createTable(tableName, STORAGE_KEY, STORAGE_VALUE);
-
+    auto table = m_memoryTableFactory->createTable(tableName, STORAGE_KEY, STORAGE_VALUE, _origin);
+    if (!table)
+    {
+        return;
```
Comment (wheatli) [inline libstoragestate/StorageState.cpp]: Print log to tell user why create table failed

File: libstorage/MemoryTable.h [hunk 2]
```diff
@@ -36,11 +36,11 @@ class MemoryTable : public Table
 
     virtual void init(const std::string& tableName);
     virtual Entries::Ptr select(const std::string& key, Condition::Ptr condition) override;
-    virtual size_t update(const std::string& key, Entry::Ptr entry, Condition::Ptr condition,
+    virtual ssize_t update(const std::string& key, Entry::Ptr entry, Condition::Ptr condition,
```
Comment (wheatli) [inline libstorage/MemoryTable.h]: use int instead of ssize_t

File: libstoragestate/StorageState.cpp (hunk 1 above)
Comment (fqliao) [reply libstoragestate/StorageState.cpp]: I will print related log in  MemoryTableFactory.

File: libstorage/MemoryTable.h (hunk 2 above)
Comment (fqliao) [reply libstorage/MemoryTable.h]: I take it.

**Pull request merged**